### PR TITLE
fix(e2e) - Change event logging to opt in to specific events only.

### PR DESCRIPTION
**Pull request by vinagesh**
## Checklist
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [ ] This pull-request is submitted against the `master` branch.

## Description of the changes
This change ensures that our E2E tests only subscribe to a specific set of event sources. After this change is checked in, I will take the changes into the aad branch as well. Having this change in master directly is beneficial and can help us upgrade the storage sdk.

### CODE REVIEW

File: e2e/test/Helpers/ConsoleEventListener.cs [hunk 1]
```diff
@@ -1,89 +1,50 @@
...
-            }
-        }
-
         protected override void OnEventSourceCreated(EventSource eventSource)
         {
-            base.OnEventSourceCreated(eventSource);
-            EnableEvents(
-                eventSource,
-                EventLevel.LogAlways
+            if (_eventFilters.Any(filter => eventSource.Name.StartsWith(filter, StringComparison.OrdinalIgnoreCase)))
```
Comment (vinagesh) [inline e2e/test/Helpers/ConsoleEventListener.cs:17]: Only subscribe to events in the filter list

File: e2e/test/Helpers/ConsoleEventListener.cs [hunk 2]
```diff
@@ -1,89 +1,50 @@
...
-            {
-                return;
-            }
-
             lock (_lock)
             {
-                if (_eventFilters.Any(ef => eventData.EventSource.Name.StartsWith(ef, StringComparison.Ordinal)))
-                {
-                    string eventIdent;
+                string eventIdent;
```
Comment (vinagesh) [inline e2e/test/Helpers/ConsoleEventListener.cs:34]: We already subscribed to a select set of events so we don't need any filtering here.

File: e2e/test/Helpers/ConsoleEventListener.cs [hunk 3]
```diff
@@ -1,89 +1,50 @@
...
+using System.Collections.Generic;
 using System.Globalization;
 using System.Linq;
 
 namespace System.Diagnostics.Tracing
 {
     public sealed class ConsoleEventListener : EventListener
     {
-        private readonly string[] _eventFilters;
+        private static readonly string[] _eventFilters = new string[] { "DotNetty-Default", "Microsoft-Azure-Devices", "Azure-Core", "Azure-Identity" };
```
Comment (vinagesh) [inline e2e/test/Helpers/ConsoleEventListener.cs:12]: Azure-Core and identity will be required later when we merge feature to here. But no harm keeping it in the list. We won't even get those events so should be a no-op.

File: e2e/test/Helpers/ConsoleEventListener.cs [hunk 4]
```diff
@@ -1,89 +1,50 @@
...
+using System.Collections.Generic;
 using System.Globalization;
 using System.Linq;
 
 namespace System.Diagnostics.Tracing
 {
     public sealed class ConsoleEventListener : EventListener
     {
-        private readonly string[] _eventFilters;
+        private static readonly string[] _eventFilters = new string[] { "DotNetty-Default", "Microsoft-Azure-Devices" };
```
Comment (abhipsaMisra) [inline e2e/test/Helpers/ConsoleEventListener.cs:12]: One suggestion is to allow the console logger to be initialized with a list of event providers. The reason for this is, we recommend customers use this class to help them debug their test code, and they wouldn't want all of our sdk relevant event sources to be written by default.

**Review by abhipsaMisra — APPROVED** [review]

File: e2e/test/Helpers/ConsoleEventListener.cs (hunk 4 above)
Comment (drwill-ms) [reply e2e/test/Helpers/ConsoleEventListener.cs:12]: Doesn't that assume customers log to ETW too?

**Review by drwill-ms — APPROVED** [review]

File: e2e/test/Helpers/ConsoleEventListener.cs (hunk 4 above)
Comment (vinagesh) [reply e2e/test/Helpers/ConsoleEventListener.cs:12]: Here is what I found on testing. OnEventSourceCreated get called even before we populate the _eventFilters in the constructor. It was extremely odd but happened. Hence, I removed that. It is possible to miss subscribing to some if we go that route. It becomes misleading. I am open to other suggestions.

**Pull request merged**